ViewSingleEvent: add CommentSection

mfocshaner opened the pull request:
added temporary button to open comment section in a separate activity.

Commit:
Comment: initial commit.

Commit:
EventModel: add comments.

Commit:
DatabaseStreamer: add comments support.

Commit:
CommentSectionFragment: initial commit.

Commit:
ViewCommentsActivity: initial commit.

Commit:
ViewSingleEvent: add button to ViewCommentsActivity.

mharsat reviewed (changes requested):
Didn't look at the code yet but ran the branch.

- The "view comments" button big and is covering the "going" section of view single event. In any design you choose, I believe the button should be small as this is not the main functionality of the screen.
- In an event that already had comments in it, It sometimes loads the comment and sometimes doesn't.
- I got a NullPointerException when attempting to add a comment to "Breakfast at Zuni's". If it is only related to mock data events then I think we can just remove those events, otherwise, it should be fixed.
- The enter new comment popup is a bit dense, and the text you write gets covered by the line below it. It should get a bit of padding from the edges of the card.
- A comment that is too long gets cut in the card. We should either set the card's size to wrap content or limit the number of characters in a comment.
- I think the commenter's name is too big and bold and so is the date and time, those could be much smaller like in Whatsapp, so we maximize the area for text.
- I love the upvote and downvote, in order for it to look the same across all devices I'd suggest to use "thumb up" and "thumb down" from the android vector assets.
- I like the different hue you gave to consecutive comments! Nice thought!

After the view single event is finished we can have a better idea of how we want to Access the comments section (having the google calendar event UI in mind).

mfocshaner commented:
> Didn't look at the code yet but ran the branch.
> 
> * The "view comments" button big and is covering the "going" section of view single event. In any design you choose, I believe the button should be small as this is not the main functionality of the screen.

Yeah, I added it that way because Ido is working on his view, so he'll add it as he sees fit. I added the button so there's access to the activity, and also so it's easy for Ido to remove it and replace it.

> * In an event that already had comments in it, It sometimes loads the comment and sometimes doesn't.

Not sure about that. It takes time to load it..
> * I got a NullPointerException when attempting to add a comment to "Breakfast at Zuni's". If it is only related to mock data events then I think we can just remove those events, otherwise, it should be fixed.

Breakfast at Zuni's a dummy event; I think Amit can remove it now he's solved the problems in his screen.
> * The enter new comment popup is a bit dense, and the text you write gets covered by the line below it. It should get a bit of padding from the edges of the card.

true
> * A comment that is too long gets cut in the card. We should either set the card's size to wrap content or limit the number of characters in a comment.

true
> * I think the commenter's name is too big and bold and so is the date and time, those could be much smaller like in Whatsapp, so we maximize the area for text.

ok, sounds alright
> * I love the upvote and downvote, in order for it to look the same across all devices I'd suggest to use "thumb up" and "thumb down" from the android vector assets.

not sure how to do that
> * I like the different hue you gave to consecutive comments! Nice thought!
thanks 😄 i thought it should have something to make it a bit unique
> 
> After the view single event is finished we can have a better idea of how we want to Access the comments section (having the google calendar event UI in mind).

In general, we don't have a lot of time until Monday's user testing; so I aimed at finishing it 80%.
Since I can't compile to my phone, it's a bit harder for me to fix the sizes of stuff, so you're more than welcome to make any visual changes you like, I trust your eye :)

Commit:
ViewCommentsActivity: fix new comment dialog.

Add padding, remove pink line, set maximal input length (100 characters).
